Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1235,10 +1386,180 @@ CGImageRef CGJPEGImageCreateFromData(NSData* data) {
+        }
+
+        colorSpaceAllocated = true;
+    }
+
+    const unsigned int numColorComponents = CGColorSpaceGetNumberOfComponents(colorSpace);
+    const unsigned int numComponents = numColorComponents + ((alphaType == kCGImageAlphaNone) ? 0 : 1);
+    const unsigned int bitsPerPixel = (bitsPerComponent == 5) ? 16 : numComponents * bitsPerComponent;
+
+    __CGSurfaceFormat format = _CGImageGetFormat(bitsPerComponent, bitsPerPixel, colorSpace, bitmapInfo);
+
+    __CGSurfaceInfo surfaceInfo = __CGSurfaceInfo(((__CGColorSpace*)colorSpace)->colorSpaceModel,
+                                                  bitmapInfo,
+                                                  bitsPerComponent,
+                                                  bitsPerPixel >> 3,
+                                                  width,
+                                                  height,
+                                                  bytesPerRow,
+                                                  data,
+                                                  format);
+
+    newImage = new CGBitmapImage(surfaceInfo);
+
+    CGContextRef ret = new __CGContext(newImage);
+    CFRelease((id)newImage);
+
+    if (colorSpaceAllocated == true) {
+        CGColorSpaceRelease(colorSpace);
+    }
+
+    return ret;
+#endif
+    return nullptr;
+}
+
+/**
+ @Status Interoperable
+*/
+CGColorSpaceRef CGBitmapContextGetColorSpace(CGContextRef context) {
+    // TODO: Consider caching colorspaceRef in CGImageRef
+    return nullptr;
+    // return (CGColorSpaceRef) new __CGColorSpace(context->Backing()->DestImage()->Backing()->ColorSpaceModel());
+}
+
+/**
+ @Status Interoperable
+*/
+size_t CGBitmapContextGetWidth(CGContextRef context) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Interoperable
+*/
+size_t CGBitmapContextGetHeight(CGContextRef context) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Interoperable
+*/
+size_t CGBitmapContextGetBytesPerRow(CGContextRef context) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Interoperable
+*/
+void* CGBitmapContextGetData(CGContextRef context) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Caveat
+ @Notes Has no copy-on-write semantics; bitmap returned is the source bitmap representing
+        the CGContext
+*/
+CGImageRef CGBitmapContextCreateImage(CGContextRef context) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+CGImageRef CGBitmapContextGetImage(CGContextRef context) {
+    if (CFGetTypeID(context) != __CGBitmapContextGetTypeID()) {
+        TraceError(TAG, L"Image requested from non-bitmap CGContext.");
+        return nullptr;
+    }
+    return ((__CGBitmapContext*)context)->_bitmapImpl.image.get();
+    // UNIMPLEMENTED();
+    // return StubReturn();
+}
+
+CGContextRef _CGBitmapContextCreateWithTexture(int width, int height, DisplayTexture* texture, DisplayTextureLocking* locking) {
```

Comment:
Indeed! I'll add a 1125 comment here too.

---
In reply to: [82922718](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r82922718) [](ancestors = 82922718)